packages/components/toggle-button/src/toggle-button.spec.ts

```diff
@@ -0,0 +1,151 @@
+      el.tabIndex = 10;
+      await el.updateComplete;
+
+      el.setAttribute('disabled', '');
+      await el.updateComplete;
+
+      el.removeAttribute('disabled');
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('tabindex', '10');
+    });
+
+    it('should be size medium', () => {
+      expect(el).to.have.attribute('size', 'md');
+    });
+
+    it('should be small size when set', async () => {
+      el.size = 'sm';
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('size', 'sm');
+    });
+
+    it('should be large size when set', async () => {
+      el.size = 'lg';
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('size', 'lg');
+    });
+
+    it('should have a default fill', () => {
+      expect(el).to.have.attribute('fill', 'ghost');
+    });
+
+    it('should not be pressed by default', () => {
+      expect(el).not.to.have.attribute('pressed');
+    });
+
+    it('should not be disabled by default', () => {
+      expect(el).not.to.have.attribute('disabled');
+      expect(el).not.to.match(':disabled');
+      expect(el.disabled).not.to.be.true;
+    });
+
+    it('should emit a click event when clicked', () => {
+      const clickEvent = new Event('click');
+      const onToggle = spy();
+      expect(el).not.to.have.attribute('pressed');
+
+      el.addEventListener('sl-toggle', onToggle);
+
+      el.dispatchEvent(clickEvent);
+
+      expect(onToggle).to.have.been.calledOnce;
+    });
+
+    it('should emit an sl-change event when pressing the enter key', async () => {
+      const onToggle = spy();
+      expect(el).not.to.have.attribute('pressed');
+
+      el.addEventListener('sl-toggle', onToggle);
+      el.focus();
+      await sendKeys({ press: 'Enter' });
+
+      expect(onToggle).to.have.been.calledOnce;
+      expect(el).to.have.attribute('pressed');
+    });
+  });
+
+  describe('disabled', () => {
+    beforeEach(async () => {
+      el = await fixture(
+        html`<sl-toggle-button disabled>
+          <sl-icon name="far-gear"></sl-icon>
+          <sl-icon name="fas-gear" slot="pressed"></sl-icon>
+        </sl-toggle-button>`
+      );
+    });
+
+    it('should have the :disabled pseudo class', async () => {
+      el.disabled = true;
+      await el.updateComplete;
+
+      expect(el).to.match(':disabled');
+    });
+
+    it('should have a tabindex of -1', async () => {
+      el.disabled = true;
+      await el.updateComplete;
+
+      expect(el).to.have.attribute('tabindex', '-1');
+    });
+
+    it('should not emit a click event when the button is disabled', async () => {
+      const clickEvent = new Event('click');
+      const preventDefaultSpy = spy(clickEvent, 'preventDefault');
+      const stopPropagationSpy = spy(clickEvent, 'stopPropagation');
+
+      el.disabled = true;
+      await el.updateComplete;
+
+      el.dispatchEvent(clickEvent);
+
+      expect(preventDefaultSpy).to.have.been.called;
```

Comment:
This is really implementation specific and brittle?. Shouldn't we just test that it doesn't emit an `sl-toggle` event?